pkg/config/validator.go

```diff
@@ -307,7 +312,48 @@ func validateSourceBindings(sl validator.StructLevel, sources map[string]Sources
 		}
 	}
 
-	validateBindPlugins(sl, enabledPluginsViaBindings)
+	validateBoundPlugins(sl, enabledPluginsViaBindings)
+	validatePluginRBAC(sl, sources, bindings)
+}
+
+func validatePluginRBAC[P providesPlugins](sl validator.StructLevel, hasPlugins map[string]P, bindings []string) {
+	// 1. identify duplicates
+	groups := make(map[string][]string)
+	for _, b := range bindings {
+		plugins := hasPlugins[b]
+		for pluginKey, plugin := range plugins.GetPlugins() {
+			if !plugin.Enabled {
+				continue
+			}
+			if g := groups[pluginKey]; len(g) == 0 {
+				groups[pluginKey] = []string{b}
+			} else {
+				groups[pluginKey] = append(groups[pluginKey], b)
+			}
```

Comment:
I believe we can append to nil slice - please verify if that works:
```suggestion
			groups[pluginKey] = append(groups[pluginKey], b)
```
Alternatively, this will be simpler:

```suggestion
			if groups[pluginKey] == nil {
				groups[pluginKey] = []string{}
			}
			groups[pluginKey] = append(groups[pluginKey], b)
```

But I believe the first option will work 👍